src/setup-beam.js

```diff
@@ -154,60 +148,46 @@ async function getOTPVersion(otpSpec0, osVersion) {
   return otpVersions.get(otpVersion) // from the reference, for download
 }
 
-async function getElixirVersion(exSpec0, otpVersion) {
+async function getElixirVersion(exSpec0, otpVersion0) {
+  const otpVersion = otpVersion0.match(/^([^-]+-)?(.+)$/)[2]
+  const otpVersionMajor = otpVersion.match(/^([^.]+).*$/)[1]
   const elixirVersions = await getElixirVersions()
   const semverVersions = Array.from(elixirVersions.keys()).sort()
-
-  const exSpec = exSpec0.match(/^v?(.+)(-otp-.+)/) || exSpec0.match(/^v?(.+)/)
-  let elixirVersion
-  if (exSpec[2] && !isStrictVersion()) {
-    throw new Error(
-      `Requested Elixir / Erlang/OTP version (${exSpec0} / ${otpVersion}) ` +
-        "should not contain '-otp-...'",
-    )
+  const exSpec = exSpec0.replace(/-otp-.*$/, '')
+  const elixirVersionFromSpec = getVersionFromSpec(exSpec, semverVersions, true)
```

Comment:
This is the only thing I don't like, the boolean blindness, but it's javascript so 🤷 😁  The only thing I could thing to do improve this is to pass around an object, I would not block on it though.